crates/bevy_picking/src/focus.rs

```diff
@@ -252,28 +317,55 @@ pub fn update_interactions(
 
 /// Merge the interaction state of this entity into the aggregated map.
 fn merge_interaction_states(
+    id: &PointerId,
     pointer_press: &PointerPress,
     hovered_entity: &Entity,
     new_interaction_state: &mut HashMap<Entity, PickingInteraction>,
 ) {
-    let new_interaction = match pointer_press.is_any_pressed() {
-        true => PickingInteraction::Pressed,
-        false => PickingInteraction::Hovered,
+    let new_interaction = if !pointer_press.is_any_pressed() {
+        PickingInteraction::Hovered
+    } else {
+        match id {
+            PointerId::Mouse => PickingInteraction::Pressed((*pointer_press).into()),
+            PointerId::Touch(_) => PickingInteraction::Pressed(PressedButtons::TOUCH),
```

Comment:
See here for the mapping from `Touch`. I think that this is the right approach for the current PR.